Assets/MixedRealityToolkit.Tests/PlayModeTests/PlayModeTestUtilities.cs

```diff
@@ -1,23 +1,82 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
+#if !WINDOWS_UWP
+// When the .NET scripting backend is enabled and C# projects are built
+// Unity doesn't include the the required assemblies (i.e. the ones below).
+// Given that the .NET backend is deprecated by Unity at this point it's we have
+// to work around this on our end.
+
 using Microsoft.MixedReality.Toolkit.Utilities;
 using Microsoft.MixedReality.Toolkit.Input;
 using UnityEngine;
+using NUnit.Framework;
+using System.Collections;
 
 namespace Microsoft.MixedReality.Toolkit.Tests
 {
     public class PlayModeTestUtilities
     {
-        public static SimulatedHandData.HandJointDataGenerator GenerateHandPose(ArticulatedHandPose.GestureId gesture, Handedness handedness, Vector3 screenPosition)
+        public static SimulatedHandData.HandJointDataGenerator GenerateHandPose(ArticulatedHandPose.GestureId gesture, Handedness handedness, Vector3 worldPosition)
         {
             return (jointsOut) =>
             {
                 ArticulatedHandPose gesturePose = ArticulatedHandPose.GetGesturePose(gesture);
                 Quaternion rotation = Quaternion.identity;
-                Vector3 position = CameraCache.Main.ScreenToWorldPoint(screenPosition);
-                gesturePose.ComputeJointPoses(handedness, rotation, position, jointsOut);
+                gesturePose.ComputeJointPoses(handedness, rotation, worldPosition, jointsOut);
             };
         }
+
+        public static IMixedRealityInputSystem GetInputSystem()
+        {
+            IMixedRealityInputSystem inputSystem;
+            MixedRealityServiceRegistry.TryGetService(out inputSystem);
+            Assert.IsNotNull(inputSystem, "MixedRealityInputSystem is null!");
+            return inputSystem;
+        }
+
+        public static InputSimulationService GetInputSimulationService()
+        {
+            IMixedRealityInputSystem inputSystem = GetInputSystem();
+            InputSimulationService inputSimulationService = (inputSystem as IMixedRealityDataProviderAccess).GetDataProvider<InputSimulationService>();
+            Assert.IsNotNull(inputSimulationService, "InputSimulationService is null!");
+            inputSimulationService.UserInputEnabled = false;
+            return inputSimulationService;
+        }
+
+        internal static IEnumerator MoveHandFromTo(Vector3 startPos, Vector3 endPos, int numSteps, ArticulatedHandPose.GestureId gestureId, Handedness handedness, InputSimulationService inputSimulationService)
```

Comment:
Could we make all these async?

Reply:
Hi Jarod, could you please explain why it might be a good idea to do that here? All of the PlayMode tests are currently just using Coroutines without async implementation, so for now I kept it using Coroutines for consistency. However, I'd like to understand why using async would be a better approach. I normally use async when I'm writing asynchronous code such as reading large files or making web / network requests, but not normally when doing playmode tests / in Unity. Perhaps I'm missing some untapped capabilities of using async instead of coroutines in Unity. Definitely willing to learn!

Reply:
> Perhaps I'm missing some untapped capabilities of using async instead of coroutines in Unity.

Just one off the top of my head, the ability to return types super easily. While that doesn't have any bearing on this situation, I think that async code in general is much easier to follow. So making these async would not be a better approach *functionally*.